script.js

```diff
@@ -1 +1,134 @@
-//😍CEOS 20기 프론트엔드 파이팅😍
+const date = document.getElementById('date');
+const todoCount = document.getElementById('todo-count');
+const todoForm = document.getElementById('todo-form');
+const todoList = document.getElementById('todo-list');
+const todoInput = document.querySelector('#todo-form input')
+
+// 날짜 함수
+function getDate() {
+  const today = new Date();
+  const year = today.getFullYear();
+  const month = today.getMonth() + 1;
+  const day = today.getDate();
+  return `⊹ ⋆ ${year}. ${month}. ${day}. ⋆ ⊹`
+}
+date.innerText = getDate();
+
+// 개수 함수
+function getTodoCount() {
+  totalTodo = todos.length;
+  doneTodo = todos.filter(todo => todo.done).length;
+  todoCount.innerText = `${doneTodo} / ${totalTodo}`;
+}
+
+// 로컬 스토리지에 저장하기, 불러오기
+let todos = [];
+
+try {
+  todos = JSON.parse(localStorage.getItem('todos')) || [];
+} catch (e) {
+  todos = [];
+}
+todos = todos.filter(todo => todo.date === getDate());
+
+function saveTodos() {
+  localStorage.setItem('todos', JSON.stringify(todos));
+}
+
+// let todos = (JSON.parse(localStorage.getItem('todos') ?? [])).filter(todo => todo.date === getDate());
```

Comment:
음 이 코드를 제가 실제로 돌려본 것은 아닌데, `JSON.parse()` 함수에 []까지 묶여있어서 그런 것이 아닐까 싶어요.
```suggestion
let todos = localStorage.getItem('todos') ? JSON.parse(localStorage.getItem('todos')) ? [].filter(todo => todo.date === getDate());
```

삼항 연산자를 활용하여 위와 같은 방식으로도 처리할 수 있지 않을까 싶어요. 

<img width="1206" alt="스크린샷 2024-09-08 오후 4 37 35" src="https://github.com/user-attachments/assets/7d37ac39-ba43-4988-b32b-0d21b3ebe395">
`JSON.parse()` 메서드의 인터페이스를 까보니, 인자로는 문자열이 오는 것이 맞아서, 오히려 '[]' 처럼 표현해줬으면 js 엔진이 알아서 배열로 파싱해줬을 수 있겠네욥 🙃